src/entity.ts

```diff
@@ -92,23 +92,23 @@ export class Entity {
         batchState.entityState = undefined;
     }
 
-    private getInput(currentRequest: RequestMessage): unknown | undefined {
+    private getInput(currentRequest: RequestMessage): T | undefined {

```

Comment:
Currently, `getInput` and `getState` are both of type `T`, which I think overly constraints the entity. What about providing two types:
one for input
one for output

I can imagine a scenario where the input to an entity is a number, and that returned value is, say, a string. Think of a fizz-buzz'ing entity :)

Reply:
Yep, the signature of `getInput` should be `<TInput>(currentRequest: RequestMessage): TInput | undefined`, not inheriting the `T` from the `Entity` signature.

`getState` should use `T` from the `Entity` signature though.